### DISCUSSION TITLE
Update MOOSE in HPC

### DISCUSSION THREAD

**Discussion by hokonkim** [idaholab/moose#18192] [Q&A General] [answered]
I used the previous MOOSE in HPC, so far.

In the HPC, MOOSE was installed as module (e.g. moose/11Oct2019).

Also, moose repository exists in my account (e.g. ~/myaccount/projects/moose).

Recently, I requested to install the latest version of MOOSE and now it was installed (moose/27Jun2021).

I think I need to update the moose repository, too.

But, I didn't update MOOSE and work in HPC so long, I forgot to how to do properly.

I checked the MOOSE website, and I think running the following commands under my account will be needed.

mkdir ~/projects
cd ~/projects
git clone https://github.com/idaholab/moose.git
cd moose
git checkout master

Also, in the website, it told me how to Compile PETSc and Compile libMesh.

I didn't remember I did compile PETSc and libMesh under my account.

I thought cloning MOOSE under my account would be enough.

As far as I know, for using the latest MOOSE, I have to....

1. load modules (module load gnu/9.3.0 openmpi/4.0.4_gnu9.3 moose/27Jun2021)
 -> there is not anaconda/2.7 anymore compared to previous MOOSE.

2. Update MOOSE repository under my account.
 -> use above "git clone" command.

3. Make application with new MOOSE, again??

I didn't run the commands, yet.

Please tell me what did I miss or what should I do to update properly.

Thanks!

**Comment by GiudGiud** [comment, marked as answer]
If your HPC people install libmesh and petsc as part of the moose module, it should be ok, and you should be able to skip step 2, unless you need a newer version. 

Are $LIBMESH_DIR and $PETSC_DIR set by the moose module?

**Reply by hokonkim** [reply]
I don't know if $LIBMESH_DIR and $PETSC_DIR set by the moose module.

Actually, I didn't remember I did run the command related to PETSc and libMesh for previous version.

Does MOOSE module in HPC work as if MOOSE repository?

I mean, under loading MOOSE module, I can recompile an application that I made.

But I don't know if it's because the MOOSE repository itself is in my account, or if it's because I've loaded the MOOSE module.

**Reply by GiudGiud** [reply]
It depends how the module was created. Is this INL HPC? If so yes.

You can examine environement variables using echo. 
So echo $MOOSE_DIR will show you if there s a moose that your application can find to built itself with
Please run this for LIBMESH_DIR and PETSC_DIR so we can see if they need to be installed. I suspect they would be part of the module though

**Reply by hokonkim** [reply]
I examine the environment like you told.

![image](https://user-images.githubusercontent.com/46009457/123691296-70a13480-d823-11eb-8d37-89349c9966b5.png)

Above image is a result after I typed several commands.

**Reply by GiudGiud** [reply]
The petsc is brought in from the openmpi module.
If there is a different one in the moose module, you will need to export that environment variable. 

You need both MOOSE_DIR and LIBMESH_DIR set properly. Please try to locate the corresponding folders in your moose module. If you cant find them, you will need to re-install libmesh and moose yourself.

**Reply by hokonkim** [reply]
I can run the MOOSE application with above given environment(old MOOSE), even I cannot locate $MOOSE_DIR or $LIBMESH_DIR.

Is this due to MOOSE repository under my account like below?

![image](https://user-images.githubusercontent.com/46009457/123699622-7c91f400-d82d-11eb-992a-0e2481540c95.png)

**Reply by hokonkim** [reply]
I summarized what I will do.

I found a previous discussion that I wrote a while ago (https://groups.google.com/g/moose-users/c/mtbblVWFRqY/m/2DQC89tgBgAJ).

It looks like I tried to compile PETSc and libMesh after cloning MOOSE in HPC, but failed (perhaps.. I didn't load module???)

Just for safety and stability....

1. (present) ermine and moose folder ->  change name as "ermine_old" and "moose_old"

2. load modules that I need (module load gnu/9.3.0 openmpi/4.0.4_gnu9.3 moose/27Jun2021)

3. cloning MOOSE under my account

4. I think I can skip compiling PETSc because it can be brought from which module I load.

5. Compile libMesh

6. Compile and Test MOOSE

What do you think? Will it work properly?

**Reply by GiudGiud** [reply]
If you are importing a moose module, you should not need to clone. You should have access to a moose somewhere, and can just set MOOSE_DIR. 
If you clone, you will have a local version of moose, and do not need a moose module UNLESS the moose module is providing you with PETSC (careful with the openmpi version then!) and libmesh. 

If this is still unclear, you should just follow the HPC instructions below and not use the moose/Jun module. 
https://mooseframework.inl.gov/getting_started/installation/hpc_install_moose.html
